File: DriftGame/libs/src/java-universal-tween-engine/tween-engine-api/src/aurelienribon/tweenengine/equations/Quart.java
package aurelienribon.tweenengine.equations;

import aurelienribon.tweenengine.TweenEquation;

/**
 * Easing equation based on Robert Penner's work: <a href="http://robertpenner.com/easing/">Look here</a>
 */
public abstract class Quart extends TweenEquation {

    public static final Quart OUT = new Quart() {
        @Override
        public float compute(float t) {
            return -((t -= 1) * t * t * t - 1);
        }

        @Override
        public String toString() {
            return "Quart.OUT";
        }
    };

}

File: DriftGame/libs/src/java-universal-tween-engine/tween-engine-api/src/aurelienribon/tweenengine/equations/Quart.kt
package aurelienribon.tweenengine.equations

import aurelienribon.tweenengine.TweenEquation

/**
 * Easing equation based on Robert Penner's work: [Look here](http://robertpenner.com/easing/)
 */
abstract class Quart : TweenEquation() {
    companion object {
        val OUT = object : Quart() {
            override fun compute(t: Float): Float {
                var t = t
                return -((1.let { t -= it; t }) * t * t * t - 1)
            }

            override fun toString() = "Quart.OUT"

        }
    }
}
